### PR TITLE
Add test for workbench example

### PR DESCRIPTION
**Pull request by frmdstryr**
Refactored the enaml_run to use https://docs.pytest.org/en/latest/monkeypatch.html and moved it to tests/utils.py

### CODE REVIEW

File: tests/test_workbench.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+#------------------------------------------------------------------------------
+# Copyright (c) 2013-2018, Nucleic Development Team.
```
Comment (MatthieuDartiailh) [inline tests/test_workbench.py]: Copyright is wrong.

**Comment by frmdstryr** [comment]
See https://travis-ci.org/nucleic/enaml/jobs/401130580 

```
Collecting QScintilla
  Downloading https://files.pythonhosted.org/packages/a6/d2/c13ab35d1747d7836d9ebdaad3bec1e7a6ce7681461f6db4765d691e2474/QScintilla-2.10.7-5.11.1-cp35.cp36.cp37.cp38-abi3-manylinux1_x86_64.whl (2.6MB)
    100% |████████████████████████████████| 2.6MB 11.2MB/s 
Collecting pyqt5==5.10.1
  Downloading https://files.pythonhosted.org/packages/e4/15/4e2e49f64884edbab6f833c6fd3add24d7938f2429aec1f2883e645d4d8f/PyQt5-5.10.1-5.10.1-cp35.cp36.cp37.cp38-abi3-manylinux1_x86_64.whl (107.8MB)
    100% |████████████████████████████████| 107.8MB 361kB/s 
Collecting sip<4.20,>=4.19.4 (from pyqt5==5.10.1)
  Downloading https://files.pythonhosted.org/packages/8a/ea/d317ce5696dda4df7c156cd60447cda22833b38106c98250eae1451f03ec/sip-4.19.8-cp36-cp36m-manylinux1_x86_64.whl (66kB)
    100% |████████████████████████████████| 71kB 28.5MB/s 
mkl-random 1.0.1 requires cython, which is not installed.
mkl-fft 1.0.0 requires cython, which is not installed.
Installing collected packages: sip, pyqt5, QScintilla
Successfully installed QScintilla-2.10.7 pyqt5-5.10.1 sip-4.19.8
```

**Comment by MatthieuDartiailh** [comment]
Indeed that is weird ! I will try to look into it (but no promise)

**Comment by frmdstryr** [comment]
This must be something with travis or how codecov merges it's report. 

When I run locally it's showing coverage, it's also in the html and xml reports, so Idk.

```bash
~/Workspace/enaml$ pytest tests/test_workbench.py -v --cov enaml | grep workbench/core
enaml/workbench/core/__init__.py                0      0   100%
enaml/workbench/core/api.py                     1      0   100%
enaml/workbench/core/command.py                 6      0   100%
enaml/workbench/core/core_plugin.py            75     18    76%
enaml/workbench/core/execution_event.py         8      0   100%

```

**Comment by MatthieuDartiailh** [comment]
I restarted the build, we will see. And I will most likely merge once Travis is done.

**Comment by MatthieuDartiailh** [comment]
I do not get what's wrong with the coverage measurement. I will merge and investigate later. Thanks again for your contribution.

**Pull request merged**